testsuite/features/step_definitions/common_steps.rb

```diff
@@ -1346,3 +1369,90 @@ def server_secret
+  raise 'error in cobbler buildiso, check the Cobbler logs' if code.nonzero?
+end
+
+When(/^I run Cobbler buildiso for distro "([^"]*)" and profile "([^"]*)" without dns entries$/) do |distro, profile|
+  tmp_dir = "/var/cache/cobbler/buildiso"
+  iso_dir = "/var/cache/cobbler"
+  _out, code = $server.run("cobbler buildiso --tempdir=#{tmp_dir} --iso #{iso_dir}/#{profile}.iso --distro=#{distro} --profile=#{profile} --exclude-dns")
+  raise 'error in cobbler buildiso, check the Cobbler logs' if code.nonzero?
+  # TODO Check for missing nameserver entry in isolinux.cfg
+  result, code = $server.run("cat #{tmp_dir}/isolinux/isolinux.cfg | grep -o nameserver")
+  # we have to fail here if the command suceeds
+  raise 'error in Cobbler buildiso, nameserver parameter found in isolinux.cfg but should not be found' if code.zero?
+end
+
+When(/^I run Cobbler buildiso "([^"]*)" for distro "([^"]*)"$/) do |param, distro|
+  # param can either be standalone or airgapped
+  # workaround to get the contents of the buildiso folder
+  step %(I run Cobbler buildiso for distro "#{distro}" and all profiles)
+  tmp_dir = "/var/cache/cobbler/buildiso"
+  iso_dir = "/var/cache/cobbler"
+  source_dir = "/var/cache/cobbler/source_#{param}"
+  $server.run("mv #{tmp_dir} #{source_dir}")
+  $server.run("mkdir -p #{tmp_dir}")
+  _out, code = $server.run("cobbler buildiso --tempdir=#{tmp_dir} --iso #{iso_dir}/#{param}.iso --distro=#{distro} --#{param} --source=#{source_dir}")
+  raise 'error in cobbler buildiso, check the Cobbler logs' if code.nonzero?
+end
+
+And(/^I check Cobbler buildiso ISO "([^"]*)" with xorriso$/) do |name|
+  tmp_dir = "/var/cache/cobbler"
+  _out, code = $server.run("cat >#{tmp_dir}/test_image <<-EOF
+BIOS
+UEFI
+EOF")
+  _out, code = $server.run("xorriso -indev #{tmp_dir}/#{name}.iso -report_el_torito 2>/dev/null | awk '/^El Torito boot img[[:space:]]+:[[:space:]]+[0-9]+[[:space:]]+[a-zA-Z]+[[:space:]]+y/{print $7}' >>#{tmp_dir}/xorriso_#{name}")
+  raise 'error while executing xorriso' if code.nonzero?
+  _out, code = $server.run("diff #{tmp_dir}/test_image #{tmp_dir}/xorriso_#{name}")
+  raise 'error in verifying Cobbler buildiso image with xorriso' if code.nonzero?
+end
+
+# item is a Cobbler item and could be distro/profile/system
+# For the parameter have a look at the Cobbler docs or man page
+Then(/^I add the Cobbler parameter "([^"]*)" with value "([^"]*)" to item "([^"]*)" with name "([^"]*)"$/) do |param, value, item, name|
+  result, code = $server.run("cobbler #{item} edit --name=#{name} --#{param}=#{value}")
+  puts("cobbler #{item} edit --name #{name} #{param}=#{value}")
+  raise 'error in adding parameter and value to Cobbler distro/profile/system' if code.nonzero?
+end
+
+# For the parameter have a look at the Cobbler docs or man page
+And(/^I check the Cobbler parameter "([^"]*)" with value "([^"]*)" in the isolinux.cfg$/) do |param, value|
```

Comment:
If only some parameters are available, maybe we want to use a regular expression with ORs.

Reply:
We can also avoid the comment

Reply:
I would prefer looking those parameters up in the docs/code since there are several ones. This makes us more independent and we do not have to touch our code if there are changes upstream.

Reply:
Keep in mind that BDD and Gherkin documentation is designed to have living documentation.
So the tests in Cucumber serves to several purposes, and one is to be attached and dependant to product code in the sense that it reflects exactly what the product code offers.

So, exposing here the available parameters, is a good strategy.
Anyway, if you prefer to keep it like that, fine for me.